### ISSUE TITLE
Error: First shared hypercore must be the same

### ISSUE THREAD

**Issue by alexindigo** [mafintosh/hypercore#107]
Not exactly clear what happened here.

Tried to start to scripts and share data between them, as shown in the example.

Got this:

```
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: First shared hypercore must be the same
    at Protocol._sameKey (/Users/alex/Projects/hyperdrive/node_modules/hypercore-protocol/index.js:404:16)
    at Protocol._onopen (/Users/alex/Projects/hyperdrive/node_modules/hypercore-protocol/index.js:272:15)
    at Protocol._onmessage (/Users/alex/Projects/hyperdrive/node_modules/hypercore-protocol/index.js:311:17)
    at Protocol._parseMessage (/Users/alex/Projects/hyperdrive/node_modules/hypercore-protocol/index.js:365:10)
    at Protocol._parse (/Users/alex/Projects/hyperdrive/node_modules/hypercore-protocol/index.js:330:37)
    at Protocol._write (/Users/alex/Projects/hyperdrive/node_modules/hypercore-protocol/index.js:238:8)
    at doWrite (/Users/alex/Projects/hyperdrive/node_modules/readable-stream/lib/_stream_writable.js:406:64)
    at writeOrBuffer (/Users/alex/Projects/hyperdrive/node_modules/readable-stream/lib/_stream_writable.js:395:5)
    at Protocol.Writable.write (/Users/alex/Projects/hyperdrive/node_modules/readable-stream/lib/_stream_writable.js:322:11)
    at Socket.ondata (_stream_readable.js:555:20)
```

Thank you.

**Comment by mafintosh** [comment]
Can you share the example you are running?

**Comment by alexindigo** [comment]
Tried to put all the details into gist – https://gist.github.com/alexindigo/96fb493345bcda1b531099cfc2facda9

Let me know if anything else is needed.

Thank you.

**Comment by mafintosh** [comment]
@alexindigo https://gist.github.com/alexindigo/96fb493345bcda1b531099cfc2facda9#file-two-js-L5 should be 'hex' instead of 'utf8' and you should use the key, not discoveryKey

**Comment by alexindigo** [comment]
I see. Thanks.
What's the difference between key and discoveryKey?

**Comment by mafintosh** [comment]
Key is the one you use to download/decrypt and verify data. Discovery key is a key derived from the normal dat but that doesnt let you decrypt or verify the data. It can be used instead to discover other peers sharing the dat without exposing the key.

**Comment by alexindigo** [comment]
Interesting. Thank you.

Btw, since you mentioned dat, some where on github I saw example of using hyperdrive directly with beaker browser, but I don't remember where it was (bumped to it deep at night and it's lost somewhere among my 1000 tabs) :)

When I tried to open `dat://` + that key in my beaker browser – it didn't load anything.
Is there some missing piece to make it available as regular dat?

Thank you.
